### PR TITLE
New random number generator for ZF3

### PR DESCRIPTION
**Pull request by ezimuel**
This is a refactor of the `Rand::randBytes()` function using the new random number generators of PHP 7, if available. For PHP < 7.0 I proposed to use the [paragonie/random_compat](https://github.com/paragonie/random_compat) project that contains an implementation of the random_bytes() for PHP >= 5.2.0. 

**Note:** this PR contains some BC breaks:
- we removed the `$strong` optional parameter in the Rand functions. Now all the random bytes/numbers generated by ZF3 will always use a secure pseudo-random number generator ([CSPRNG](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator)).
- we removed the return false in `Rand::getBytes($length)` if `$length <= 0`;
- we changed the behaviour of the `Exception` for `Rand::getBytes()` and `Rand::getInteger()`.

### CODE REVIEW

File: src/Rand.php [hunk 1]
```diff
@@ -43,58 +42,26 @@ public static function getBytes($length, $strong = false)
...
-     *
-     * @return RandomLib\Generator
-     */
-    public static function getAlternativeGenerator()
-    {
-        if (null !== static::$generator) {
-            return static::$generator;
-        }
-        if (!class_exists('RandomLib\\Factory')) {
+        require_once 'vendor/paragonie/random_compat/lib/random.php';
```
Comment (Ocramius) [inline src/Rand.php]: Already included: https://github.com/paragonie/random_compat/blob/b3bbb09d8457e9dfc51194594689f71803c48eaf/composer.json#L33

Comment (ezimuel) [inline src/Rand.php]: @Ocramius thanks, I didn't notice this. I need to refactor the code than.

**Comment by ezimuel** [comment]
@Ocramius I removed the include and refactor the Exception behaviour adding more tests.

**Comment by ezimuel** [comment]
@paragonie-scott We used your [random_compat](https://github.com/paragonie/random_compat) project in our new random number generator for ZF3. It will be awesome if you can review the code of `Rand::getBytes()` and `Rand::getInteger()` that use your functions for PHP < 7. Thanks!

**Comment by paragonie-scott** [comment]
https://github.com/ezimuel/zend-math/blob/feature/new-rand-zf3/src/Rand.php#L144-L150

This has the potential to introduce bias if `$listLen` is not an even multiple of 2. I'd suggest, instead:

``` php
        $result = '';
        for ($i = 0; $i < $length; $i++) {
            $pos     = random_int(0, $listLen - 1);
            $result .= $charlist[$pos];
        }
```

**Comment by ezimuel** [comment]
@paragonie-scott good catch, thanks! For those who are wondering why the modulo operator (%) can introduce a bias in randomness, [read this answer](http://crypto.stackexchange.com/a/22769).
I'll use the `static::getInteger()` instead of `random_int`.

**Comment by paragonie-scott** [comment]
Latest commit looks good to me. :)

**Comment by ezimuel** [comment]
@weierophinney this PR is ready to be merged in develop for ver 3.0.0, thanks!

**Pull request merged**